### PR TITLE
Add pivot examples to vignette

### PR DESCRIPTION
**Pull request by grantmcdermott**
I also cleaned up a few typos and inconsistencies.

### CODE REVIEW

File: vignettes/polars.Rmd [hunk 1]
```diff
@@ -299,6 +302,53 @@ dat$groupby(
...
+# indo_wide$melt(id_vars = "time") # default column names are "variable" and "value"
+indo_wide$melt(id_vars = "time", variable_name = "subject", value_name = "conc") 
+```
+
+Basic functionality aside, it should be noted that `pivot` can peform
+aggregations as part of the reshaping operation. This is useful when you have
+multiple observations per ID variable that need to be collapsed into unique
+values. The aggregating functions can be arbritrarily complex, but let's
+consider a relatively simple example using our `dat` ("mtcars") DataFrame from
+earlier: What is the median MPG value (`mpg`) across cylinders (`cyl`), cut by
```
Comment (etiennebacher) [inline vignettes/polars.Rmd:336]:
```suggestion
earlier: what is the median MPG value (`mpg`) across cylinders (`cyl`), cut by
```

**Comment by grantmcdermott** [comment]
@etiennebacher Thanks for the review. I think this should be good to go once you're happy (or we can wait until someone else to take a look too).

File: vignettes/polars.Rmd [hunk 2]
```diff
@@ -299,6 +302,53 @@ dat$groupby(
...
+
+```{r}
+# indo_wide$melt(id_vars = "time") # default column names are "variable" and "value"
+indo_wide$melt(id_vars = "time", variable_name = "subject", value_name = "conc") 
+```
+
+Basic functionality aside, it should be noted that `pivot` can peform
+aggregations as part of the reshaping operation. This is useful when you have
+multiple observations per ID variable that need to be collapsed into unique
+values. The aggregating functions can be arbritrarily complex, but let's
```
Comment (etiennebacher) [inline vignettes/polars.Rmd:334]:
```suggestion
values. The aggregating functions can be arbitrarily complex, but let's
```

**Review by etiennebacher — APPROVED** [review]
LGTM, thanks @grantmcdermott

**Pull request merged**